src/java/org/apache/cassandra/repair/state/ParticipateState.java

```diff
@@ -148,16 +200,43 @@ public void accept()
         }
     }
 
-    public static class Job extends AbstractState<Job.State, RepairJobDesc>
+    public static class Job extends AbstractState<Job.State, RepairJobDesc> implements WeightedHierarchy.InternalNode
     {
         public enum State { ACCEPT, SNAPSHOT, VALIDATION, SYNC }
 
+        private final ParticipateState participateState;
         private final AtomicReference<ValidationState> validation = new AtomicReference<>(null);
         private final ConcurrentMap<SyncState.Id, SyncState> syncs = new ConcurrentHashMap<>();
+        private static final long EMPTY_SIZE;
+        static
+        {
+            PrepareMessage msg = new PrepareMessage(nextTimeUUID(), Collections.emptyList(), Collections.emptyList(), false, 0L, false, PreviewKind.NONE);
+            ParticipateState participateState = new ParticipateState(Clock.Global.clock(), FBUtilities.getBroadcastAddressAndPort(), msg);
+            RepairJobDesc desc = new RepairJobDesc(null, null, null, null, null);
+            EMPTY_SIZE = ObjectSizes.measure(new Job(Clock.Global.clock(), participateState, desc));
```

Comment:
```suggestion
            EMPTY_SIZE = ObjectSizes.measure(new Job(Clock.Global.clock(), null, null));
```

could cleanup some code / imports...